### ISSUE TITLE
Make html-minifier compatible with knockout comments.

### ISSUE THREAD

**Issue by salvador-dali** [kangax/html-minifier#145]
Knockout (popular MVVM library) has specific syntax which looks like comments, but in fact it represents logic. These comments are easily recognizable (have specific pattern), but because they are still comments, they are stripped out by html-minifier.

Knowing that knockout is used by a large community, it would be nice to have something like isKnockout: true, to remove all comments, but leave knockout specific comments.

A more detailed exlanation can be found http://stackoverflow.com/q/22544914/1090562

**Comment by kangax** [comment]
I'm thinking to add an option `ignoreKoComments: { true | false }` which would ignore any comments starting with `<!-- ko ... -->` and ending with `<!-- /ko -->`. Does that sound right?

**Comment by nikolajtesla** [comment]
Yes, based on my knowledge and documentation knockout comments can look only like this. Thank you for your effort to make tool cooler.

**Comment by stevenvachon** [comment]
How about creating an option for custom comments to ignore? There are many libraries out there..

**Comment by kangax** [comment]
Yes, that might make more sense. How do you imagine it working though? Via regex?

**Comment by stevenvachon** [comment]
Yep, in an object/array-like syntax:

``` js
ignoreCustomComments: [
    {
        before: /<!--\s+ko[^"-->"]+-->/,
        after: /<!--\s+\/ko\s+-->/
    }
]
```

or with `<!--` and `-->` concatenated in later:

``` js
ignoreCustomComments: [
    {
        before: /\s+ko[^"-->"]+/,
        after: /\s+\/ko\s+/
    }
]
```

**Comment by kangax** [comment]
Yep, looks good /cc @salvador-dali @nikolajtesla

**Comment by JBlackCat** [comment]
This would be very helpful for me since I am using Apache SSIs and still need to strip other comments without removing the SSIs.

**Comment by blowsie** [comment]
Could anyhow here show me how I can utilise this to prevent my copyright information being removed?

Thanks 

```
  <!--
    <%= pkg.name %> - <%= pkg.version %> - <%= pkg.buildID %> - <%= grunt.template.today("yyyy-mm-dd") %>
    <%= pkg.homepage %>
    Copyright (c) <%= grunt.template.today("yyyy") %> <%= pkg.author.name %>
  -->
```
